### PR TITLE
[Kubernetes Testing] Add DUT Join Kubernetes Master Test Cases 

### PR DESCRIPTION
**Pull request by isabelmsft**
### Description of PR

Summary:

Added 4 SONiC DUT Join Test Cases, as described [here](https://github.com/isabelmsft/sonic-mgmt/blob/k8s_test_doc/docs/K8s-test-plan.md)

TC_JOIN_2: Ensure kube server disable flag works as expected- DUT joins master when disable=false, and DUT resets from master when disable=true
TC_JOIN_3: Ensure DUT joins after config reload with no config change (config saved with disable=false)
TC_JOIN_4: Ensure DUT resets from master when config reload toggles disable to true (config saved with disable=true)
TC_JOIN_5: Ensure DUT joins master when config reload toggles disable to false (config saved with disable=false)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

#### What is the motivation for this PR?
We want to ensure that the SONiC DUT joins and resets from Kubernetes master as expected under various conditions

#### How did you do it?
Used `k8scluster` fixture to retrieve VIP, used `duthost` to configure DUT based on test steps in the testplan linked above. 

#### How did you verify/test it?

```
isl@a08d5cc274e9:/data/sonic-mgmt/tests$ ./run_tests.sh -d vlab-01 -n vms-kvm-t0 -f ../ansible/vtestbed.csv -i ../ansible/veos_vtb,../ansible/k8s_ubuntu_vtb -u -b k8s_vms1_21 -c k8s/test_config_reload.py
=== Running tests in groups ===
/usr/local/lib/python2.7/dist-packages/ansible/parsing/vault/__init__.py:44: CryptographyDeprecationWarning: Python 2 is no longer supported by the Python core team. Support for it is now deprecated in cryptography, and will be removed in a future release.
  from cryptography.exceptions import InvalidSignature
========================================================================================= test session starts ==========================================================================================
platform linux2 -- Python 2.7.17, pytest-4.6.5, py-1.9.0, pluggy-0.13.1
ansible: 2.8.12
rootdir: /data/sonic-mgmt/tests, inifile: pytest.ini
plugins: html-1.22.1, forked-1.3.0, xdist-1.28.0, metadata-1.10.0, repeat-0.8.0, ansible-2.2.2
collected 3 items                                                                                                                                                                                      

k8s/test_config_reload.py::test_config_reload_no_toggle PASSED                                                                                                                                   [ 33%]
k8s/test_config_reload.py::test_config_reload_toggle_join PASSED                                                                                                                                 [ 66%]
k8s/test_config_reload.py::test_config_reload_toggle_reset PASSED                                                                                                                                [100%]

------------------------------------------------------------------------ generated xml file: /data/sonic-mgmt/tests/logs/tr.xml ------------------------------------------------------------------------
====================================================================================== 3 passed in 546.76 seconds ======================================================================================
isl@a08d5cc274e9:/data/sonic-mgmt/tests$ 
```

```
========================================================================================= test session starts ==========================================================================================
platform linux2 -- Python 2.7.17, pytest-4.6.5, py-1.9.0, pluggy-0.13.1
ansible: 2.8.12
rootdir: /data/sonic-mgmt/tests, inifile: pytest.ini
plugins: html-1.22.1, forked-1.3.0, xdist-1.28.0, metadata-1.10.0, repeat-0.8.0, ansible-2.2.2
collected 1 item                                                                                                                                                                                       

k8s/test_disable_flag.py::test_disable_flag PASSED                                                                                                                                               [100%]

------------------------------------------------------------------------ generated xml file: /data/sonic-mgmt/tests/logs/tr.xml ------------------------------------------------------------------------
====================================================================================== 1 passed in 159.43 seconds ======================================================================================
isl@a08d5cc274e9:/data/sonic-mgmt/tests$ 
```

### Documentation 
https://github.com/isabelmsft/sonic-mgmt/blob/k8s_test_doc/docs/K8s-test-plan.md

### CODE REVIEW

**Comment by isabelmsft** [comment]
retest vsimage please

File: tests/k8s/test_config_reload.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
+import pytest
+import time
+import k8s_test_utilities as ku
+
+from tests.common.helpers.assertions import pytest_assert
+from tests.common.platform.processes_utils import wait_critical_processes
+
+
+
+
```
Comment (wangxin) [inline tests/k8s/test_config_reload.py:10]: Two extra blank lines.

Need topology mark like:
```
pytestmark = [
    pytest.mark.topology('any')
]
```

File: tests/k8s/test_disable_flag.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import pytest
+import time
+import k8s_test_utilities as ku
+
+from tests.common.helpers.assertions import pytest_assert
+
+
+
```
Comment (wangxin) [inline tests/k8s/test_disable_flag.py:8]: One extra blank line.

Need topology mark like:
```
pytestmark = [
    pytest.mark.topology('any')
]
```

File: tests/k8s/test_config_reload.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    Performs config reload
+
+    Ensures that DUT is still joined to master after config reload
+
+    Args:
+        duthost: DUT host object
+        k8scluster: shortcut fixture for getting cluster of Kubernetes master hosts
+    """
+    master_vip = k8scluster.get_master_vip()
+    ku.join_master(duthost, master_vip) # Assertion within to ensure successful join
```
Comment (wangxin) [inline tests/k8s/test_config_reload.py:26]: Can you make `master_vip` a property of `K8sMasterCluster`? For example:
```
class K8sMasterCluster(object):

    @property
    def vip(self):
        return self.haproxy.mgmt_ip
```
Then the above code can be simpler:
```
ku.join_master(duthost, k8scluster.vip)
```

File: tests/k8s/test_join_available_master.py [hunk 4]
```diff
@@ -1,13 +1,11 @@
 import pytest
-import logging
 import time
 import k8s_test_utilities as ku
 
 from tests.common.helpers.assertions import pytest_assert
 
 WAIT_FOR_SYNC = 60
 
-logger = logging.getLogger(__name__)
```
Comment (wangxin) [inline tests/k8s/test_join_available_master.py:9]: Need topology mark like:
```
pytestmark = [
    pytest.mark.topology('any')
]
```

File: tests/k8s/test_config_reload.py (hunk 1 above)
Comment (qiluo-msft) [inline tests/k8s/test_config_reload.py:10]: Remove extra empty lines

File: tests/k8s/test_config_reload.py [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+    ku.join_master(duthost, master_vip) 
+
+    duthost.shell('sudo config reload -y')
+    wait_critical_processes(duthost)
+
+    server_connect_exp_status = False
+    server_connect_act_status = ku.check_connected(duthost)
+    server_connect_status_updated = ku.poll_for_status_change(duthost, server_connect_exp_status)
+    pytest_assert(server_connect_status_updated, "Unexpected k8s server connection status after config reload, Expected server connected status: {}, Found server connected status: {}".format(server_connect_exp_status, server_connect_act_status))
+
```
Comment (qiluo-msft) [inline tests/k8s/test_config_reload.py:105]: Remove tail empty lines

**Review by qiluo-msft — COMMENTED** [review]
As comments

File: tests/k8s/test_config_reload.py (hunk 1 above)
Comment (isabelmsft) [reply tests/k8s/test_config_reload.py:11]: Fixed

File: tests/k8s/test_disable_flag.py (hunk 2 above)
Comment (isabelmsft) [reply tests/k8s/test_disable_flag.py:10]: Fixed spacing and added marker

File: tests/k8s/test_config_reload.py (hunk 3 above)
Comment (isabelmsft) [reply tests/k8s/test_config_reload.py]: Yes, thank you for the suggestion

File: tests/k8s/test_config_reload.py (hunk 1 above)
Comment (isabelmsft) [reply tests/k8s/test_config_reload.py:11]: Removed

File: tests/k8s/test_join_available_master.py (hunk 4 above)
Comment (isabelmsft) [reply tests/k8s/test_join_available_master.py]: Added marker

File: tests/k8s/test_config_reload.py (hunk 5 above)
Comment (isabelmsft) [reply tests/k8s/test_config_reload.py]: Removed

**Comment by wangxin** [comment]
Change `class K8sMasterCluster():` to `class K8sMasterCluster(object):` should can address the LGTM alert '1 for Property in old-style class'

**Review by wangxin — APPROVED** [review]

**Review by qiluo-msft — APPROVED** [review]

**Review by renukamanavalan — APPROVED** [review]

**Pull request merged**